babel/board.js

```diff
@@ -104,11 +138,16 @@ export class Board {
     }
 
     needsShuffle() {
-
+        return !this.hasMatch();
     }
 
     hasMatch() {
-        return !this.needsShuffle();
+        let chunks = iterchunks(this.orbs);
+        // [[[1, 2, 3], [2, 3, 4]], [[3, 4, 5], [4, 5, 6]]] becomes
+        //  [[1, 2, 3], [2, 3, 4], [3, 4, 5], [4, 5, 6]]
+        let flatChunks = _.flattenDepth(chunks, 1);
+        let hasWideStyleMatch = _.some(_.map(flatChunks, hasMatchInSingleRow));
+        return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
```

Comment:
I like that it is split up into single row and pair of rows matches. It must be much faster to just test the simple way first and return true as soon as it finds one, and then go through the more complicated to search for matches if necessary.